### PR TITLE
Send list via AMQP

### LINKED ISSUE #17: Send 'list' message to message queue
- [ ] Replace proto_fetch to use rabbitmq
- [ ] Listen to exchange devices.list
- [ ] Send device.cmd.list

### PR DESCRIPTION
**Pull request by ramonhpr**
This PR closes #17 

How to test:
install pika python package
run: test/mock-connector.py
run knotd: sudo src/knotd -nr -c src/knotd.conf

Expected result:
the mock-connector should show the messages in json
And knotd should add the dbus device (mocked) in d-feet

### CODE REVIEW

File: src/msg.c [hunk 1]
```diff
@@ -154,13 +154,39 @@ static void mydevice_free(void *data)
...
 	l_free(mydevice);
 }
 
+static void schema_dup_foreach(void *data, void *user_data)
+{
+	struct l_queue *schema = user_data;
+	knot_msg_schema *msg = data;
+	knot_msg_schema *msg_dup = l_memdup(msg, sizeof(*msg));
+
+	l_queue_push_tail(schema, msg_dup);
```
Comment (vitbaq) [inline src/msg.c]: Put l_memdup(msg, sizeof(*msg) on l_queue_push_tail or check if msg_dup exist.

File: src/cloud.c [hunk 2]
```diff
@@ -166,6 +173,34 @@ static bool on_cloud_receive_message(const char *exchange,
...
+/**
+ * cloud_list_devices:
+ *
+ * Requests cloud to list the devices from the gateway.
+ * The confirmation that the cloud received the message comes from a callback
+ * set in function cloud_set_read_handler with message type LIST_MSG.
+ *
+ * Returns: 0 if successful and a KNoT error otherwise.
+ */
+int cloud_list_devices(void)
```
Comment (vitbaq) [inline src/cloud.c]: Put cloud_list_devices after unregister function

File: src/cloud.c [hunk 3]
```diff
@@ -166,6 +196,44 @@ static bool on_cloud_receive_message(const char *exchange,
...
+{
+	amqp_bytes_t queue_cloud;
+	json_object *jobj_empty;
+	const char *json_str;
```
[the remark this answers is not in the document]
Comment (ramonhpr) [reply src/cloud.c]: The jobj_empty is clearer that I will send a string "{}"

File: src/msg.c (hunk 1 above)
Comment (ramonhpr) [reply src/msg.c]: Done

**Pull request merged**